Notes on the Four-Fifths Rule

riedgar-ms opened the pull request:
<!--- If relevant, make sure to add a description of your changes in docs/user_guide/installation_and_version_guide/v<major>.<minor>.<patch>.rst -->

## Description
<!--- Provide a general summary of your changes. -->
<!--- Mention related issues, pull requests, or discussions with #<issue/PR/discussion ID>. -->
<!--- Tag people for whom this PR may be of interest using @<username>. -->

This adds a short discussion of the "Four-Fifths Rule" to the "Fairness in Machine Learning" section of the user guide. It is an explanation of why users probably don't want to use the rule as their fairness criterion.

This replaces #1042 , which was getting old and had a bunch of merge conflicts.


## Tests
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation
<!--- Select all that apply. -->
- [ ] no documentation changes needed
- [x] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

## Screenshots
<!--- If your PR makes changes to visualizations (e.g., matplotlib code) or the website, please include a screenshot of the result. -->

riedgar-ms commented:
This doesn't seem to be appearing on the docs? And building locally is note working but not giving me any sort of error :-(

riedgar-ms commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+Typically, the rule is implemented by using the demographic parity ratio introduced
+in the :ref:`disparity_metrics` section, with a result considered 'fair' if the ratio
+exceeds 80% for all identified subgroups.
+*Unthinking application of this threshold is wrong in many scenarios.*
```
Fair enough. Have adjusted.

LeJit commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+relevant control population.
+A causal link between the statistic and the alleged discrimination must also
+be demonstrated.
+However, the 'Four-Fifths Rule' does not prove or disprove any case.
```
Does 'Four-Fifths Rule' need to be capitalized?

hildeweerts replied:
I assume "prove"/"disprove" in the legal sense of the word?

LeJit commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+be demonstrated.
+However, the 'Four-Fifths Rule' does not prove or disprove any case.
+It is possible to argue successfully that a demographic parity
+ratio of 50% is still fair.
```
This may be better explained with an example (e.g. the fire department exam case).

hildeweerts replied:
+1 to adding more examples (same for the 95%)

riedgar-ms replied:
Can you remind me of that, please? Do you mean:
https://en.wikipedia.org/wiki/Ricci_v._DeStefano

LeJit replied:
Yes. This is the correct legal case. Although now that I re-read the case, I think it would be a better example of the opposite idea: How reducing the disparate impact required an illegal intervention.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
 the algorithm may impact the intended outcomes of a given model.
 
+
+The 4/5ths Rule: A Common Trap
```
As we really want people to read this, perhaps we should come up with a slightly more click-baity title, like "Why you should (almost) never apply the 4/5ths rule". 

My personal preference would go to something even more general, e.g., something like: "How 'fair' should my model be?". This is a common question data scientists have, and with this phrasing they might be more likely to read the text even if they've never heard of the 4/5ths rule. (This would also provide a nice space to add more guidelines on how to translate harms into fairness metrics, in accordance with [learning objective 1.6](https://github.com/orgs/fairlearn/projects/3/views/1))

Additionally, instead of introducing this as a separate trap, an alternative position of this paragraph would be right after the explanation of the fairness constraints. In a way, blindly applying the 4/5ths rule could be seen as an example of the formalism trap and/or a portability trap. Instead of introducing it as a separate trap, we could refer to it as an example of (one of) those traps.

riedgar-ms replied:
Clickbaity would be something like "You'll never guess what 90% of fairness experts say about the 4/5ths rule!!!"

I've moved it up in the doc, but am still mulling a title.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+
+In the literature around fairness in machine learning, one will often find
+the so-called "4/5ths rule" or "80% rule" used to assess whether a model
+(or mitigation technique) has produced a 'fair' result.
```
I would add a line here that says that this 4/5th rule originates from US labor law.

riedgar-ms replied:
Mentioned (slightly further down).... of course, calling it 'labor' implicitly means it's 'US' ;-)

hildeweerts replied:
The intended the emphasis was on *here*! Mostly related to my next comment re. differentiation between legal definitions of discrimination and "fairness".

hildeweerts replied:
PS. In an unexpected turn of events there are legal systems outside of the Anglosphere (what?!), but I suppose "arbeidsrecht" won't be very understandable for most of our readers ;)

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+In the literature around fairness in machine learning, one will often find
+the so-called "4/5ths rule" or "80% rule" used to assess whether a model
+(or mitigation technique) has produced a 'fair' result.
+Typically, the rule is implemented by using the demographic parity ratio introduced
```
I would emphasize that this is the interpretation/translation computer scientists/fairness researchers have given to the rule.

This might be a good place to emphasize the difference between fairness and illegal discrimination.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+*Application of this threshold is wrong in many scenarios.*
+
+**What exactly does the rule mean?**
+Legally, the 80% threshold is a limit for
```
I would prefer something like "In US labor law" to emphasize that this is specific to the US.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+`prima facie evidence <https://en.wikipedia.org/wiki/Prima_facie>`_
+that illegal discrimination has occurred relative to a chosen,
+relevant control population.
+A causal link between the statistic and the alleged discrimination must also
```
From this sentence it is not really clear to me what a "A causal link between the statistic and the alleged discrimination" would look like - perhaps an example would help.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+It is equally possible to argue that even at a 95% ratio, illegal
+discrimination has occurred.
+Conflating the *disparate impact metric* with the *legal finding of disparate impact*
+is a form of epistemic trespassing.
```
In my opinion, "epistemic trespassing", sounds too formal for a user guide, especially without further explanation on what we mean by it (case in point: if you Google the term you mostly find academic papers).

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+Conflating the *disparate impact metric* with the *legal finding of disparate impact*
+is a form of epistemic trespassing.
+
+**Where is the 4/5ths rule valid?**
```
Nitpick: something like "In which case" might be a bit more encompassing than "where" (which mostly implies location to me)

riedgar-ms replied:
Have removed this subhead

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+**Where is the 4/5ths rule valid?**
+The answer is that it is only valid within a specific area of US
+federal employment law.
+Outside this context, the rule has no particular claim to validity.
```
Nitpick: "having no particular claim to validity" sounds very formal

riedgar-ms replied:
Simplified

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+
+The 4/5ths Rule: A Common Trap
+------------------------------
+
```
Maybe add some introductory sentence(s) here that says that a common question data scientists have is what value their fairness metrics should be for the model to be fair.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+
+**What exactly does the rule mean?**
+Legally, the 80% threshold is a limit for
+`prima facie evidence <https://en.wikipedia.org/wiki/Prima_facie>`_
```
Maybe add a small sentence to explain what prima facie evidence is.

riedgar-ms replied:
That's what the link is for.

hildeweerts replied:
In my opinion we should avoid jargon as much as possible, and if we use at least explain it in the text if we do. I'm probably spending too much time on such a detail, but we've seen in other texts (e.g., on construct validity) that using a lot of (new) terminology can make texts less accessible for our primary audience (data scientists). I would definitely count "prima facie" as jargon, but that's perhaps because I'm a non-native English speaker. But it's not a hill I'm willing to die on so if others disagree we can keep it in the current form.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+**What exactly does the rule mean?**
+Legally, the 80% threshold is a limit for
+`prima facie evidence <https://en.wikipedia.org/wiki/Prima_facie>`_
+that illegal discrimination has occurred relative to a chosen,
```
Is the word "chosen" particularly relevant here?

riedgar-ms replied:
Removed

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+be demonstrated.
+However, the 'Four-Fifths Rule' does not prove or disprove any case.
+It is possible to argue successfully that a demographic parity
+ratio of 50% is still fair.
```
We need to make sure that we clearly distinguish between "fairness" and "illegal discrimination" in this paragraph.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+ratio of 50% is still fair.
+It is equally possible to argue that even at a 95% ratio, illegal
+discrimination has occurred.
+Conflating the *disparate impact metric* with the *legal finding of disparate impact*
```
As we typically do not say "disparate impact metric", perhaps we should say something like:
"Conflating the *demographic parity ratio*, sometimes referred to as *disparate impact*, with the *legal finding of disparate impact*". Or already mention this earlier in the text, e.g., when we first refer to the demographic parity ratio.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+
+Taken together, we see that applying the 4/5ths rule will
+not be appropriate in most cases.
+Even in cases where it is appropriate, it does not automatically
```
"Appropriate" sounds a bit odd in this sentence (because we actually say that it it isn't in the next sentence). Maybe something like:
```suggestion
Even in cases where the rule is applicable, it does not automatically
```

riedgar-ms replied:
Fixed.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+Taken together, we see that applying the 4/5ths rule will
+not be appropriate in most cases.
+Even in cases where it is appropriate, it does not automatically
+avoid legal jeopardy, much less ensure that results are fair.
```
I like this sentence a lot! I would be great if we could emphasize the difference between fairness and illegal discrimination more explicitly earlier on in the text.

hildeweerts commented on docs/user_guide/fairness_in_machine_learning.rst:
```diff
@@ -435,6 +435,49 @@ incorporate the processes associated with crime reporting, the offense-trial
+not be appropriate in most cases.
+Even in cases where it is appropriate, it does not automatically
+avoid legal jeopardy, much less ensure that results are fair.
+
```
I think we can add to this paragraph that this type of situation is why we try to avoid legal terminology as much as possible in Fairlearn, which explains why we do not call it "disparate impact" but "demographic parity".

riedgar-ms replied:
Done.

hildeweerts reviewed (commented):
I love that we are adding this!!!

riedgar-ms commented:
@hildeweerts  and @LeJit I've pushed another draft, which takes on some of your comments, but also changes enough that addresses your comments but not quite directly (since I've done some restructuring of the text). As such, I've left several of your comments open so that we have them for reference.

Also, can you confirm that I had the right case for the firefighters, please?
